e2e/testdata/fn-render/fn-failure-output-no-truncate/.expected/config.yaml

```diff
@@ -0,0 +1,22 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# One of the functions in the pipeline fails resulting in
+# non-zero exit code and no changes in the resources.
+exitCode: 1
+disableOutputTruncate: true
+stdErr: "Invalid type. Expected: [integer,null], given: string in object 'apps/v1/Deployment//nginx-deployment' in file resources.yaml in field spec.replicas"
```

Comment:
I commented in one of the issues that separating `stderr` and `stdOut` seems like unnecessary complexity. What's the use case for it?

In terms of testing: I think we should have a few e2e tests in this repo to test the presentation layer. But otherwise, we should only test the structured output which should provide all the information we want to verify. In particular, packages in the catalog repo shouldn't test the presentation layer. Otherwise, you'd get brittle change detector tests:

https://testing.googleblog.com/2015/01/testing-on-toilet-change-detector-tests.html

Changing the presentation layer in the future should involve updating just a few tests.

Reply:
@mengqiy FYI

Reply:
Agree. `stdOut` and `stdErr` are optional in the test. They can be safely removed if we don't want to check them.

Reply:
Agree

Reply:
Limiting the presentation related testing to a few test cases make sense.